janus_server/src/aggregator.rs

```diff
@@ -230,20 +364,22 @@ fn aggregator_filter<C: 'static + Clock>(
     let upload_endpoint = warp::path("upload")
         .and(warp::post())
         .and(with_cloned_value(aggregator))
-        .and(with_decoded_message())
-        .and_then(|aggregator: Aggregator<C>, report: Report| async move {
+        .and_then(|aggregator: Aggregator<C>| async {
             // Only the leader supports upload
             if aggregator.role != Role::Leader {
                 return Err(warp::reject::not_found());
             }
-
-            aggregator
-                .handle_upload(&report)
-                .await
-                .map_err(warp::reject::custom)?;
-
-            Ok(reply::with_status(warp::reply(), StatusCode::OK)) as Result<_, Rejection>
+            Ok(aggregator)
         })
+        .and(with_decoded_message())
+        .then(
+            |aggregator: Aggregator<C>, report_res: Result<Report, Error>| async move {
+                aggregator.handle_upload(&report_res?).await?;
+
+                Ok(warp::reply()) as Result<_, Error>
```

Comment:
I prefer to explicitly set status HTTP 200 rather than rely on the implicit behavior of `warp::reply`, because then we can point to an obvious place where we implement the RFC requirement. Plus it'll make this consistent with other endpoints that return HTTP statuses besides OK.